doc/nrf/ug_radio_coex.rst

```diff
@@ -0,0 +1,196 @@
+
+   The ``*childImageName*`` string must be one of the following values:
+
+   *  ``multiprotocol_rpmsg`` for multiprotocol applications having support for both 802.15.4 and Bluetooth.
+   *  ``802154_rpmsg`` for applications having support for 802.15.4, but not for Bluetooth.
+   *  ``hci_rpmsg`` for application having support for Bluetooth, but not for 802.15.4.
+
+
+
+WiFi Coexistence implementations
+********************************
+
+The following CX implementations are available:
+
+* :ref:`Generic 3-pin interface`
+* :ref:`"Thread" implementaion`
+
+Each implementaion is a plugin consisting of one or more C source files interfacing with the MPSL API (which in turn communicates with the
+SR protocol drivers).
+
+When either CX variant is enabled, the driver requests access to the RF medium from the PTA and informs it about the operation it's about
+to perform. It also reacts properly to the information from the PTA that access to RF medium was granted/denied.
+
+Generic 3-pin interface
+=======================
+
+This implementation uses the following pins to communicate with the PTA:
+
+* REQUEST pin - output controlled by the protocol driver, it is asserted in order to request RF access from the PTA. It can be configured by
+  setting the `req-gpios` property of the `nrf_radio_coex` device tree node.
+* PRIORITY pin - output controlled by the protocol driver, it contains the information about the the type of operation
+  (RX/TX) to perform. It can be configured by setting the `pri-dir-gpios` property of the `nrf_radio_coex` device tree node.
+* GRANT pin - output controlled by the PTA, asserts, when PTA grants access to RF to the 802.15.4 and deasserts when it denies the access. It can
+  be configured by setting the `grant-gpios` property of the `nrf_radio_coex` device tree node.
+
+The support for this interface is provided in a single-file plugin located in the sdk-nrf repo, in the subsys/mpsl/cx directory.
+It is only available to use with the Zephyr RTOS.
+
+Adding support for CX generic 3-pin interface
+---------------------------------------------
+
+An out of the box supported type of PTA is the so called "generic 3-pin interface".
+
+To use the generic 3-pin interface PTA, complete the following steps:
+
+1. Set the :kconfig:`CONFIG_MPSL_CX_GENERIC_3PIN` Kconfig option to ``y``.
+
+2. Add the device tree `nrf_radio_coex` node in the way as described in :ref:`Hardware description`.
+
+After completing these steps no other function calls in the application are needed and the Coex will work.
+
+
+"Thread" implementaion
+======================
+
+This implementation is similar to the :ref:`Generic 3-pin interface`, however it is the REQUEST pin which contains information about the
+transfer direction and it is not multiplexed by using the PRIORITY pin (the device tree parameter `pri-dir-gpios` name is misleading here) This
+implementation uses the following pins to communicate with the PTA:
```

Comment:
This is not true.
See here https://github.com/nrfconnect/sdk-nrf/blob/main/subsys/mpsl/cx/thread/mpsl_cx_thread.c#L123

To be on safe side I would simply write 

```This implementation is similar to the :ref:`Generic 3-pin interface`, however PRIORITY pin behavior is different. It uses the following pins to communicate with the PTA:```

Reply:
Wrong.
Do you see line: https://github.com/nrfconnect/sdk-nrf/blob/main/subsys/mpsl/cx/thread/mpsl_cx_thread.c#L128
REQUEST is active when ops is either TX or RX (or both, why not). 
This is *exactly* the same for "generic 3pin"
https://github.com/nrfconnect/sdk-nrf/blob/main/subsys/mpsl/cx/generic_3pin/mpsl_cx_generic_3pin.c#L181
So I see no justification for writing in doc for "thread" anything else than for "generic 3pin".

You wrote `REQUEST pin which contains information about the transfer direction`.
As we know that REQUEST is 1 if and only if ops contains TX or RX flag or both,  if you were outside and observed REQUEST is 1 how would you distinguish between TX or RX? That's why I see this `contains information about transfer direction` problematic.

In fact there is only one difference between "3-pin" and "thread", it is how the PRIORITY is controlled.

Reply:
I got confused in some moment, changed